### ISSUE TITLE
Upgrade from 2.1.12 to 2.1.23

### ISSUE THREAD

**Issue by Ricardo-Custodio** [nilsteampassnet/TeamPass#985]
Hello,

I'am upgrading from teampass version 2.1.12 to 2.1.23, but in step 5 the page is in "please wait..." state forever after click on Launch button.
The issue is similar to #548, but with the error: "Undefined variable: abs_path in /var/www/html/teampass/install/upgrade.php on line 432"
On the previous steps, everything worked fine. 

My settings.php from version 2.1.12 does not have any reference to sk.php file, since it is used only on teampass version 2.1.13 and above.
Also, after click on Launch button the sk.php file is not created on location indicated on Absolute path do SaltKey.

![teampass-step5](https://cloud.githubusercontent.com/assets/12646960/7865609/e8bdf37c-0561-11e5-85fb-a00a2395d441.png)

**Comment by nilsteampassnet** [comment]
Can you please provide the error log from Apache?
Try following those [instructions](http://teampass.net/usage/how-to-communicate-an-error-log.html).

**Comment by Ricardo-Custodio** [comment]
This is solved, apparently it was a permission setting in the folder 'includes'. At the first time the folder was not on chmod 777 and after correcting it, step 5 launch button was not able to confirm that change.

FYI, after all steps completed with success, the sk.php was created, but was missing the salk key, so it appears like:
@define('SALT', ''); //Never Change it once it has been used !!!!!
So I had to paste the current salt key on sk.php from my old setting.php file

I also had a problem using LDAP for login fixed on #878, and I had to correct all the rules on 'rule management' page because they were mixed up.

Now everything is working.
Thank you,
